txnkv/transaction/pessimistic.go

```diff
@@ -86,6 +86,7 @@ func (actionPessimisticRollback) tiKVTxnRegionsNumHistogram() prometheus.Observe
 func (action actionPessimisticLock) handleSingleBatch(c *twoPhaseCommitter, bo *retry.Backoffer, batch batchMutations) error {
 	m := batch.mutations
 	mutations := make([]*kvrpcpb.Mutation, m.Len())
+	c.txn.GetMemBuffer().RLock()
```

Comment:
If the contention happens here would it be possible that `actionPessimisticLock` tasks are processed one by one according to the [feature](https://pkg.go.dev/sync#RWMutex) of the `RWLock`? I'm not quite sure if it may introduce extra risks.

Reply:
Do you mean that all later `RLock` requests will wait for a queuing write lock request? Could you point out which write lock may cause it to run one by one?

Reply:
For example:
```
Update  // Try to acquire the write lock.
    Some possible parallel executors(like parallel projection in the early issue)
              selectLockExec // Try to acquire read locks.
```

The update execution may try to `Set` new key-value pairs into the memory buffer which needs to acquire the write lock, while the parallel `actionPessimisticLock` operations triggered by `LockKeys` are being processed. If an `actionPessimisticLock` gets the read lock successfully first and then the write lock acquisition is pending, other `actionPessimisticLock` requests may have to wait.

Reply:
Got it. Yet I think it will have little impact if all these operations are fast. And the goroutines that are blocked after the write lock will then acquire the read lock at the same time, so they can run concurrently as before.

I will try to design a worst case to see how much impact it will bring.